filters/openpolicyagent/serveresponsewithregopolicy/serveresponsewithregopolicy.go

```diff
@@ -0,0 +1,119 @@
+package serveresponsewithregopolicy
+
+import (
+	"time"
+
+	"github.com/zalando/skipper/filters"
+	"gopkg.in/yaml.v2"
+
+	"github.com/zalando/skipper/filters/openpolicyagent"
+	"github.com/zalando/skipper/filters/openpolicyagent/internal/envoy"
+	"github.com/zalando/skipper/filters/openpolicyagent/internal/util"
+)
+
+type Spec struct {
+	factory    openpolicyagent.OpenPolicyAgentFactory
+	configOpts []func(*openpolicyagent.OpenPolicyAgentInstanceConfig) error
+}
+
+func NewServeResponseWithRegoPolicySpec(factory openpolicyagent.OpenPolicyAgentFactory, opts ...func(*openpolicyagent.OpenPolicyAgentInstanceConfig) error) Spec {
+	return Spec{
+		factory:    factory,
+		configOpts: opts,
+	}
+}
+
+func (s Spec) Name() string {
+	return "serveResponseWithRegoPolicy"
+}
+
+const (
+	paramBundleName int = iota
+	paramEnvoyContextExtensions
+)
+
+func (s Spec) CreateFilter(config []interface{}) (filters.Filter, error) {
+	var err error
+
+	sargs, err := util.GetStrings(config)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(sargs) < 1 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	if len(sargs) > 2 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	bundleName := sargs[paramBundleName]
```

Comment:
The same here as in the other filter, do not use const but explicit 0 and 1 to index sargs.